### PR TITLE
EOS-19888: Setup Consul - Basic service

### PR DESCRIPTION
**Pull request by tanujashinde0405**

### CODE REVIEW

File: srv/components/misc_pkgs/consul/files/consul_server.json.j2 [hunk 1]
```diff
@@ -3,32 +3,52 @@
...
+          {% if not loop.last %}
+          "{{ node }}",
+          {% else %}
+          "{{ node }}"
+          {% endif %}
+        {% endfor -%}
+     ]
 }
 
+
```
Comment (ypise) [inline srv/components/misc_pkgs/consul/files/consul_server.json.j2:54]: This extra empty line is unnecessary at the end of file.

File: srv/components/misc_pkgs/consul/teardown.sls [hunk 2]
```diff
@@ -33,40 +33,40 @@ include:
...
-# Remove Consul user:
-#   user.absent:
-#     - name: consul
+Reload service daemons post consul-agent.service removal:
+   cmd.run:
+     - name: systemctl daemon-reload
 
+Remove Consul user:
+   user.absent:
+     - name: consul
```
Comment (ypise) [inline srv/components/misc_pkgs/consul/teardown.sls:68]: Is this not automatically effected by rpm uninstall?

File: pillar/components/firewall.sls [hunk 3]
```diff
@@ -35,3 +35,7 @@ firewall:
       uds:
         - 5000/tcp
         - 5000/udp
+      consul:
```
Comment (ypise) [inline pillar/components/firewall.sls:38]: Consul is supposed to run on private data interface. Comment these entries for now.

File: api/python/provisioner/srv/salt/firewall/config.sls [hunk 4]
```diff
@@ -25,6 +25,7 @@ saltmaster:
     - ports:
       - 4505/tcp
       - 4506/tcp
+      - 8301/tcp
```
Comment (ypise) [inline api/python/provisioner/srv/salt/firewall/config.sls:28]: Consul is supposed to run on private data interface. Comment this entry for now.

**Review by ypise — CHANGES_REQUESTED** [review]

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**